crates/vm/levm/src/execution_handlers.rs

```
@@ -154,32 +154,29 @@ impl<'a> VM<'a> {
 
             let code_length_u64: u64 = code_length
                 .try_into()
-                .map_err(|_| VMError::Internal(InternalError::ConversionError))?;
+                .map_err(|_| InternalError::TypeConversion)?;
 
-            let code_deposit_cost: u64 =
-                code_length_u64
-                    .checked_mul(CODE_DEPOSIT_COST)
-                    .ok_or(VMError::Internal(
-                        InternalError::ArithmeticOperationOverflow,
-                    ))?;
+            let code_deposit_cost: u64 = code_length_u64
+                .checked_mul(CODE_DEPOSIT_COST)
+                .ok_or(InternalError::Overflow)?;
 
             // Revert
             // If the first byte of code is 0xef
             // If the code_length > MAX_CODE_SIZE
             // If current_consumed_gas + code_deposit_cost > gas_limit
             let validate_create = if code_length > MAX_CODE_SIZE {
-                Err(VMError::ContractOutputTooBig)
+                Err(ExceptionalHalt::ContractOutputTooBig)
             } else if contract_code
                 .first()
                 .is_some_and(|val| val == &INVALID_CONTRACT_PREFIX)
             {
-                Err(VMError::InvalidContractPrefix)
+                Err(ExceptionalHalt::InvalidContractPrefix)
             } else if self
                 .current_call_frame_mut()?
                 .increase_consumed_gas(code_deposit_cost)
                 .is_err()
             {
-                Err(VMError::OutOfGas(OutOfGasError::MaxGasLimitExceeded))
+                Err(ExceptionalHalt::OutOfGas)
```

I think this could be simpler with this:
```
} else if let Err(e) = self
    .current_call_frame_mut()?
    .increase_consumed_gas(code_deposit_cost)
{
    return Err(e);
} else {
```

---

Here we don't want to return the error. We want to assign it to a variable so that it then generates the `ExecutionReport` with it.